fix(replay): Fix missing canvas docs for browser frameworks (#9005)
getsentry/sentry-docs: docs/platforms/javascript/common/session-replay/index.mdx

<Alert level="">
  Canvas recording is currently in beta. Please submit an issue on
  [GitHub](https://github.com/getsentry/sentry-javascript) if you encounter any
  bugs.
</Alert>

<Alert level="warning">
  There is currently no PII scrubbing in canvas recordings!
</Alert>

If you want to record HTML canvas elements, you'll need to add an additional integration in your Sentry configuration. The canvas integration is exported from the browser SDK, so no additional package is required. Canvas recording is opt-in and will be tree-shaken from your bundle if it's not being used:

<PlatformContent includePath="session-replay/setup-canvas" />

#### 3D and WebGL Canvases

The canvas recording integration works by exporting the canvas as an image (at a rate of 2 frames per second). However, in order to export images from 3D and WebGL canvases, the integration needs to enable `preserveDrawingBuffer` which can negatively affect canvas performance. If your canvas application is impacted by enabling `preserveDrawingBuffer`, you'll need to enable manual snapshotting and call a `snapshot()` method inside of your re-paint loop. There are two steps to using manual snapshotting:

> **Step 1.**
> Enable manual snapshotting when initializing the `ReplayCanvas` integration.

Reviewer: ```suggestion
1. Enable manual snapshotting when initializing the `ReplayCanvas` integration.
```

I don't think we really format steps like this anywhere else in the docs, so for simplicity & consistency, let's just use numbered lists for step by step instructions.

> ```javascript
> Sentry.replayCanvasIntegration({
>   // Enabling the following will ensure your canvas elements are not forced
>   // into `preserveDrawingBuffer`.
>   enableManualSnapshot: true,
> });
> ```

Reviewer: ```suggestion
  ```javascript
  Sentry.replayCanvasIntegration({
    // Enabling the following will ensure your canvas elements are not forced
    // into `preserveDrawingBuffer`.
    enableManualSnapshot: true,
  });
  ```
```

Indent so it gets tabbed under the steps

> **Step 2**
> Call the the following `snapshot()` method inside your application's paint loop. `snapshot()` needs to be called in the same execution loop as the canvas draw commands, otherwise you may be snapshotting empty canvas buffers. This is due to how WebGL works when `preserveDrawingBuffer` is `false`.
> 
> ```javascript
> function paint() {
>   const canvasRef = document.querySelector("#my-canvas");
>   Sentry.getClient().getIntegrationByName("ReplayCanvas").snapshot(canvasRef);
> }
> ```

Reviewer: ```suggestion
1. Call the following `snapshot()` method inside your application's paint loop. `snapshot()` needs to be called in the same execution loop as the canvas draw commands; otherwise, you may be snapshotting empty canvas buffers. This is due to how WebGL works when `preserveDrawingBuffer` is `false`.
  
  ```javascript
  function paint() {
    const canvasRef = document.querySelector("#my-canvas");
    Sentry.getClient().getIntegrationByName("ReplayCanvas").snapshot(canvasRef);
  }
  ```
```

<PlatformContent includePath="session-replay/setup" />

> <PlatformSection supported={["javascript"]} notSupported={["javascript.deno", "javascript.bun", "javascript.wasm", "javascript.capacitor", "javascript.electron"]}>

Reviewer 2: Do we need to add the `supported` part here? 🤔  or would it be enough to add the excludes?

Author: Ah yeah removed, this means that only javascript browser is supported (and not frameworks).

### Canvas Recording

<Alert level="">
  Canvas recording is currently in beta. Please submit an issue on
  [GitHub](https://github.com/getsentry/sentry-javascript) if you encounter any
  bugs.
</Alert>

<Alert level="warning">
  There is currently no PII scrubbing in canvas recordings!
</Alert>

If you want to record HTML canvas elements, you'll need to add an additional integration in your Sentry configuration. The canvas integration is exported from the browser SDK, so no additional package is required. Canvas recording is opt-in and will be tree-shaken from your bundle if it's not being used:

> <PlatformContent includePath="session-replay/setup-canvas" />

Author: Not sure if this is better? This limits the amount of copy & pasted text by only moving the code snippets to this platform content include.
